pom.xml

```diff
@@ -2136,12 +2138,66 @@
+                <maven.plugin.surefire.argLine>-XX:+IgnoreUnrecognizedVMOptions
+                    --add-opens=java.base/java.lang=ALL-UNNAMED
+                    --add-opens=java.base/java.lang.invoke=ALL-UNNAMED
+                    --add-opens=java.base/java.lang.reflect=ALL-UNNAMED
+                    --add-opens=java.base/java.io=ALL-UNNAMED
+                    --add-opens=java.base/java.net=ALL-UNNAMED
+                    --add-opens=java.base/java.nio=ALL-UNNAMED
+                    --add-opens=java.base/java.util=ALL-UNNAMED
+                    --add-opens=java.base/java.util.concurrent=ALL-UNNAMED
+                    --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED
+                    --add-opens=java.base/sun.nio.ch=ALL-UNNAMED
+                    --add-opens=java.base/sun.nio.cs=ALL-UNNAMED
+                    --add-opens=java.base/sun.security.action=ALL-UNNAMED
+                    --add-opens=java.base/sun.security.tools.keytool=ALL-UNNAMED
+                    --add-opens=java.base/sun.security.x509=ALL-UNNAMED
+                    --add-opens=java.base/sun.util.calendar=ALL-UNNAMED
+                    -Djdk.reflect.useDirectMethodHandle=false
+                    -Dio.netty.tryReflectionSetAccessible=true</maven.plugin.surefire.argLine>
+            </properties>
+            <build>
+                <pluginManagement>
+                    <plugins>
+                        <plugin>
+                            <groupId>com.diffplug.spotless</groupId>
+                            <artifactId>spotless-maven-plugin</artifactId>
+                            <version>2.43.0</version>
+                            <configuration>
+                                <java>
+                                    <googleJavaFormat>
+                                        <version>1.22.0</version>
+                                    </googleJavaFormat>
+                                </java>
+                                <!-- The current version of spotless(2.30.0) and google-java-format(1.7) do not support Java 21.
+                                     To run on Java 21, the 'spotless:check' needs to be skipped for the moment.
+                                     Re-evaluate once support for Java 8 is dropped. -->
+                                <skip>true</skip>
+                            </configuration>
+                        </plugin>
+                    </plugins>
+                </pluginManagement>
+            </build>
+        </profile>
+
         <!-- For development only, not generally applicable for all modules -->
         <profile>
             <id>scala-2.13</id>
             <properties>
                 <scala.binary.version>2.13</scala.binary.version>
-                <scala.version>2.13.8</scala.version>
+                <!-- Scala version above 2.13.11 support Java 21. -->
+                <scala.version>2.13.11</scala.version>
```

Comment:
`maven.plugin.silencer.version` might be required to update accordingly. anyway, let's do this upgrading independently.